src/main/java/com/google/sps/servlets/utils/Statistics.java

```diff
@@ -15,93 +15,140 @@
-        BigDecimal scorePoint = BigDecimal.valueOf(sentimentScores.get(scoreIdx));
-        if ((scorePoint.compareTo(nextPoint) < 0) || nextPoint.compareTo(UPPER_END) == 0) {
-          aggregateValues.put(currentRange, aggregateValues.get(currentRange) + 1);
+      for (scoreIdx = updatingScoreIdx; scoreIdx < userCommentList.size(); scoreIdx++) {
+        BigDecimal scorePoint = BigDecimal.valueOf(userCommentList.get(scoreIdx).getScore());
+        if ((scorePoint.compareTo(nextPoint) < 0) || nextPoint.compareTo(upperEnd) == 0) {
+          currentFrequency += 1;
+          // TODO: add topNumComments to the priority queue highMagnitudeList
         } else {
           break;
         }
       }
       // update the score pointer
       updatingScoreIdx = scoreIdx;
+      sentimentBucketList.add(
+          new SentimentBucket(
+              convertQueueToList(highMagnitudeComments), currentFrequency, currentRange));
+    }
+    return sentimentBucketList;
+  }
+
+  private ArrayList convertQueueToList(PriorityQueue<UserComment> inputQueue) {
+    ArrayList<UserComment> returnList = new ArrayList<>();
+    while (!inputQueue.isEmpty()) {
+      returnList.add(inputQueue.poll());
     }
+    return returnList;
+  }
+
+  /**
+   * Set the average value of given sentiment values.
+   *
+   * @param userCommentList a list of userComment to calculate the average score for
+   * @return the average score of userCommentList
+   */
+  private double getAverageScore(List<UserComment> userCommentList) {
+    return userCommentList.stream()
+        .mapToDouble(UserComment::getScore)
+        .average()
+        .orElseThrow(
+            () ->
+                new RuntimeException("Unable to calculate average score due to empty input list."));
   }
 
   /**
-   * Set the average score of given sentiment scores. Throws an runtime exception if the average
-   * score is not valid or none of the sentiment scores is valid.
+   * Set the average value of given sentiment magnitude.
    *
-   * @param sentimentScores a list of score values from -1.0 to 1.0
+   * @param userCommentList a list of userComment to calculate the average magnitude for
+   * @return the average magnitude of userCommentList
    */
-  private void setAverageScore(List<Double> sentimentScores) throws RuntimeException {
-    averageScore =
-        sentimentScores.stream()
-            .mapToDouble(i -> i)
-            .average()
-            .orElseThrow(
-                () ->
-                    new RuntimeException(
-                        "Unable to calculate sentiment average due to empty input list."));
+  private double getAverageMagnitude(List<UserComment> userCommentList) {
+    return userCommentList.stream()
+        .mapToDouble(UserComment::getMagnitude)
+        .average()
+        .orElseThrow(
+            () ->
+                new RuntimeException(
+                    "Unable to calculate average magnitude due to empty input list."));
```

Comment:
These two methods are almost identical except for the input. Can you pass in the field you want as a parameter?